packages/core/contracts/financial-templates/optimistic-distributor/OptimisticDistributor.sol

```diff
@@ -84,8 +84,10 @@ contract OptimisticDistributor is Lockable, MultiCaller, Testable {
     // Merkle Distributor can be set only once.
     MerkleDistributor public merkleDistributor;
 
-    // Interface parameters that can be synced and stored in the contract.
-    OptimisticOracleInterface public optimisticOracle;
+    // Optimistic Oracle address is fetched at deployment and not upgradable from this contract.
+    // Any Optimistic Oracle upgrades would require redeployment of this contract while previosly funded rewards'
+    // distribution can still be verified through old Optimistic Oracle instance.
+    OptimisticOracleInterface public immutable optimisticOracle;
```

Comment:
This is totally fine, but we should recognize the implications. If there is something wrong with the Optimistic Oracle or the DVM, this contract would not be able to upgrade away from that broken contract. Further, if the UMA DVM is upgraded to an interface that no longer supports this old Optimistic Oracle implementation, that would break this contract.

The alternative, as pointed out in the audit, is that this contract has to deal with mid-request changes in the OO.

What's the plan for upgrades for _this system_? Is this contract able to be deprecated easily?